src/components/ChatEntry.js

```diff
@@ -1,22 +1,45 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
 
 const ChatEntry = (props) => {
+  const source = props.sender === 'Vladimir' ? 'local' : 'remote';
+  const [liked, setLiked] = useState(props.liked);
+  const toggleLikeButton = () => {
+    const heart = document.getElementById(props.id);
+    setLiked(!liked);
+    if (liked) {
+      heart.textContent = '🤍';
+      props.onUpdate(-1);
+    } else {
+      heart.textContent = '❤️';
+      props.onUpdate(1);
+    }
+    setLiked(!liked);
+  };
```

Comment:
Ah, though looking at this, this is a bit of an awkward fallout from the decision above.

To make sure the heart button updates the setting, you have to use `useState` here to keep track of the liked status. But I notice that your if statement is "backwards" here, and that's because I figure you realized that `setLiked` does not update the `liked` variable immediately. Instead these setter functions that `useState` returns updates on re-render.

And there's nothing "wrong" per se with the logic being backwards here, but it is something that could be a bit of confusion for sure.

I think the best solution would be that rather than always drawing the empty heart and letting it change when the button is clicked, pick which heart to draw based on `props.liked`. Then you can simply call `setLiked` here and it will be updated on re-render to the right heart.

Reply:
Also there's an extra call to `setLiked`, which I presume is a typo, and from testing my clone of your code, you can remove it.